frontend/src/search/search.js

```diff
@@ -0,0 +1,315 @@
+        super(props);
+        this.state = {
+            loading: true,
+            data: null,
+            isAdvanced: false,
+            searchedText: '',
+            tagData: null,
+            photographerData: null,
+        };
+    }
+
+    updateSearchData = (searchData) => {
+        this.setState({ ...searchData });
+    }
+
+    async componentDidMount() {
+        try {
+            const searchTagResponse = await fetch('/api/get_tags/');
+            const searchTags = await searchTagResponse.json();
+            const { tags, photographers } = searchTags;
+            // Sort by name and then by number, if the photographers have one
+            photographers.sort((a, b) => {
+                const aName = a.name;
+                const bName = b.name;
+                if (aName && bName) {
+                    if (a.number && b.number) {
+                        return aName.localeCompare(bName);
+                    }
+                    if (a.number) {
+                        return -1;
+                    }
+                    if (b.number) {
+                        return 1;
+                    }
+                    return aName.localeCompare(bName);
+                }
+                if (aName) {
+                    return -1;
+                }
+                if (bName) {
+                    return 1;
+                }
+                return 0;
+            });
+            this.setState({ photographerData: photographers, tagData: tags, loading: false });
+        } catch (e) {
+            console.log(e);
+        }
+    }
+
+    // This follows the full text + advanced search model here: http://photogrammar.yale.edu/search/
+    render() {
+        if (!this.state.tagData || !this.state.photographerData) {
+            return (<LoadingPage/>);
+        }
+        return (
+            <>
+                <Navbar />
+                <div className='search-page page row'>
+                    <div className='col-sm-12 col-lg-4 search-form'>
+                        <h1>Search</h1>
+                        <SearchForm
+                            updateSearchData={this.updateSearchData}
+                            tagData={this.state.tagData}
+                            photographerData={this.state.photographerData}
+                        />
+                    </div>
+                    <div className='col-sm-12 col-lg-8'>
+                        {
+                            this.state.data
+                            && <div>
+                                <h2>
+                                    {this.state.isAdvanced
+                                        ? 'Advanced Search Results'
+                                        : 'Search Results'}
+                                </h2>
+                                <p>{this.state.searchText}</p>
+                                <div className='search-results'>
+                                    {this.state.data.map((photo, k) => {
```

Comment:
Cleanup TODO (not preventing merge) -- I'd like to see this factored out into a component for readability.